ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveMaterializedViewRule.java

```diff
@@ -74,25 +74,21 @@
               HiveJoinProjectTransposeRule.LEFT_PROJECT,
               HiveJoinProjectTransposeRule.RIGHT_PROJECT,
               HiveProjectMergeRule.INSTANCE))
-      .addRuleInstance(ProjectRemoveRule.INSTANCE)
+      .addRuleInstance(ProjectRemoveRule.Config.DEFAULT.toRule())
       .addRuleInstance(HiveRootJoinProjectInsert.INSTANCE)
       .build();
 
   public static final MaterializedViewProjectFilterRule INSTANCE_PROJECT_FILTER =
-      new MaterializedViewProjectFilterRule(HiveRelFactories.HIVE_BUILDER,
-          true, PROGRAM, false);
+      MaterializedViewProjectFilterRule.Config.DEFAULT.toRule();
```

Comment:
I think we should pass the `HIVE_BUILDER` in the rule. Moreover we should double-check that the other parameters of the rule (`generateUnionRewritting`, `fastBailOut`, etc.) are set correctly. The comment applies also to the rules below.